packages/calypso-e2e/src/lib/pages/published/published-post-page.ts

```diff
@@ -77,6 +82,9 @@ export class PublishedPostPage extends BaseContainer {
 	async likePost(): Promise< void > {
 		await this._click( selectors.likeButton );
 		await this.frame.waitForSelector( selectors.likedText, { state: 'visible' } );
+		const attributes = await this.frame.getAttribute( selectors.likeButton, 'class' );
+		const isLiked = attributes?.includes( 'liked' );
+		assert.strictEqual( isLiked, true );
```

Comment:
Assertions should probably be in the test, not the page object.  Do we need a new method to return if the post is liked and assert that in the test?

Reply:
I would avoid using assertions in helpers as well. In this specific case it looks like the `await this.frame.waitForSelector( selectors.likedText, { state: 'visible' } );` would be enough to verify the like happened, no?

Reply:
We use assertions [here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts#L78) and [here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/components/support-component.ts#L174) as two examples. For some of them, the assertions were originally in the tests but was moved into the objects as part of ensuring the action took place.

In this specific case I probably don't need to rely on assertions. @WunderBart the wait alone is not enough as the Post Like/Unlike button is particularly finicky and there needs to be additional verification to ensure that intended action did take place, hence the proposed change.